\begin{savequote}[75mm]
“I didn’t want to just know names of things. I remember really wanting to know how it all worked.”
\qauthor{Elizabeth Blackburn}
\end{savequote}

\chapter{Somatic Variant Detection}\label{chapter:virtualnormal}
\setcounter{figure}{-1}
\setcounter{table}{-1}
\setcounter{section}{-1}

\begin{figure}[t!]
\includegraphics[height=10em]{frontmatter/images/chapter-header-variants-tools.png}
\end{figure}
\setcounter{figure}{-1}
\setcounter{table}{-1}
\setcounter{section}{-1}


Sequencing of tumours often involves the co-sequencing of a sample of healthy tissue from the same individual. This allows us to distinguish the tumour-specific variants from those present in the germline of the patient. When such a matching normal sample is not available, the analysis of variants becomes more complex. In this chapter, we investigate whether a \emph{virtual normal} may be used in lieu of an associated normal in such cases. A virtual normal consists of a large set of samples from healthy, unrelated, genetically diverse individuals.

As a first step in this analysis, we integrated a suite of tools for variant analysis and visualisation into Galaxy (CGtag). We then used these tools to compare perfomance of a virtual normal as an alternative to a matched tumour-normal pair.

This chapter contains the following sub-chapters:

\begin{enumerate}[label=\ref{chapter:virtualnormal}.\arabic*]
\itemsep-0.5em
\setcounter{enumi}{-1}
\item \textbf{CGtag:} Complete Genomics Toolkit and Annotation in a Cloud-based Galaxy
\item Discriminating somatic and germline mutations in tumor DNA samples without matching normals
\end{enumerate}
