\cleartorightpage
\begin{savequote}[75mm]
“I didn’t want to just know names of things. I remember really wanting to know how it all worked.”
\qauthor{Elizabeth Blackburn}
\end{savequote}

\chapter{Somatic Variant Detection}\label{chapter:virtualnormal}
\setcounter{figure}{-1}
\setcounter{table}{-1}
\setcounter{section}{-1}

\begin{figure}[t!]
\includegraphics[height=10em]{frontmatter/images/chapter-header-variants-tools.png}
\end{figure}
\setcounter{figure}{-1}
\setcounter{table}{-1}
\setcounter{section}{-1}


Sequencing of tumours often involves the co-sequencing of a sample of healthy tissue from the same individual. This allows us to distinguish the tumour-specific variants from those present in the germline of the patient. When such a matching normal sample is not available, the analysis of variants becomes more complex. In this chapter, we investigate whether a \emph{virtual normal} may be used in lieu of an associated normal in such cases. A virtual normal consists of a large set of samples from healthy, unrelated, genetically diverse individuals.

As a first step in this analysis, we integrated a suite of tools for variant analysis and visualisation into Galaxy (CGtag). We then used these tools to compare performance of a virtual normal as an alternative to a matched tumour-normal pair.

This chapter contains the following sub-chapters:

\begin{enumerate}[label=\ref{chapter:virtualnormal}.\arabic*]
\itemsep-0.5em
\setcounter{enumi}{-1}
\item \textbf{CGtag: Complete Genomics Toolkit and Annotation in a Cloud-based Galaxy}
Different sequencing techniques are constantly being developed, and often come with custom analysis tools. This was also the case for Complete Genomics (CG) data. CG defined it's own data format and supplied specialized analysis tools, optimized for their own sequencing data. These were command-line tools, and in order to increase their accessibility, I \emph{wrapped} them (made available) in Galaxy. Furthermore, to increase interoperability of the CG tools and datasets, I created some additional tools for file format conversion, to allow other, non-CG tools to be applied to CG datasets. Note that while this publication mentions availability in the CTMM-TraIT Tool Shed, the tools have since moved to the main Galaxy Tool Shed. While the tools are still available for analysis or historic data, the tools have become obsolete as the sequencing techniques have evolved.

\item \textbf{Discriminating somatic and germline mutations in tumor DNA samples without matching normals}. In this work, I used CG-sequenced tumour-normal pairs and CG analysis tools to evaluate the utility of using as large set of samples from diverse, healthy and unrelated individuals, to serve as a \emph{virtual normal sample}, in absence of the commonly used \emph{associated normal} sample originating from healthy tissue from the same patient. As with the previous sub-chapter, the methods and code developed for this work are still available (from the main Galaxy Tool Shed rather than the CTMM-TraIT Galaxy), but have become somewhat obsolete now, since they were geared specifically towards CG-datasets. The approach is still valid, but would require some re-implementation to work for today's sequencing data. Such is often the nature of modern bioinformatics; the high speed of technological advances in the sequencing techniques leaves a high turnover of analysis tools in its wake.


\end{enumerate}
