\begin{savequote}[75mm]
Nulla facilisi. In vel sem. Morbi id urna in diam dignissim feugiat. Proin molestie tortor eu velit. Aliquam erat volutpat. Nullam ultrices, diam tempus vulputate egestas, eros pede varius leo.
\qauthor{Quoteauthor Lastname}
\end{savequote}

\chapter{Technical: iReport, myFAIR, Circos?, Training?}
\setcounter{figure}{-1}
\setcounter{table}{-1}
\setcounter{section}{-1}


\newthought{There's something to be said} for having a good opening line. Morbi commodo, ipsum sed pharetra gravida, orci  $x = 1/\alpha$ magna rhoncus neque, id pulvinar odio lorem non turpis \cite{Eigen1971, Knuth1968}. Nullam sit amet enim. Suspendisse id velit vitae ligula volutpat condimentum. Aliquam erat volutpat. Sed quis velit. Nulla facilisi. Nulla libero. Vivamus pharetra posuere sapien. Nam consectetuer. Sed aliquam, nunc eget euismod ullamcorper, lectus nunc ullamcorper orci, fermentum bibendum enim nibh eget ipsum. Donec porttitor ligula eu dolor. Maecenas vitae nulla consequat libero cursus venenatis. Nam magna enim, accumsan eu, blandit sed, blandit a, eros.
$$\zeta = \frac{1039}{\pi}$$

\verb+short intro to chapter tying the different papers together +

% For an example of a full page figure, see Fig.~\ref{fig:myFullPageFigure}.
\newpage

%%%%%%%%%%%%%%%%%
%    iReport
%%%%%%%%%%%%%%%%%

\articletitle{iReport: A generalised Galaxy solution for integrated experimental reporting}
Saskia Hiltemann\textsuperscript{\ref{affil:emc-bioinf},\ref{affil:emc-urology}},
Youri Hoogstrate\textsuperscript{\ref{affil:emc-bioinf},\ref{affil:emc-urology}},
Peter van der Spek\textsuperscript{\ref{affil:emc-bioinf}},
Guido Jenster\textsuperscript{\ref{affil:emc-urology}},
Andrew Stubbs\textsuperscript{\ref{affil:emc-bioinf}}

\small
\begin{enumerate}
\itemsep-0.5em
\item Department of Bioinformatics, Erasmus Medical Center, Rotterdam, The Netherlands. \label{affil:emc-bioinf}
\item Department of Urology, Erasmus Medical Center, Rotterdam, The Netherlands. \label{affil:emc-urology}
\end{enumerate}
\normalsize


Published in: GigaScience \\
DOI: \url{https://doi.org/10.1186/2047-217X-3-19} \\

\section*{Abstract}

\textbf{Background:} Galaxy offers a number of visualisation options with components, such as Trackster, Circster and Galaxy Charts, but currently lacks the ability to easily combine outputs from different tools into a single view or report. A number of tools produce HTML reports as output in order to combine the various output files from a single tool; however, this requires programming and knowledge of HTML, and the reports must be custom-made for each new tool.\\
\textbf{Findings:}We have developed a generic and flexible reporting tool for Galaxy, iReport, that allows users to create interactive HTML reports directly from the Galaxy UI, with the ability to combine an arbitrary number of outputs from any number of different tools. Content can be organised into different tabs, and interactivity can be added to components. To demonstrate the capability of iReport we provide two publically available examples, the first is an iReport explaining about iReports, created for, and using content from the recent Galaxy Community Conference 2014. The second is a genetic report based on a trio analysis to determine candidate pathogenic variants which uses our previously developed Galaxy toolset for whole-genome NGS analysis, CGtag. These reports may be adapted for outputs from any sequencing platform and any results, such as omics data, non-high throughput results and clinical variables.\\
\textbf{Conclusions:} iReport provides a secure, collaborative, and flexible web-based reporting system that is compatible with Galaxy (and non-Galaxy) generated content. We demonstrate its value with a real-life example of reporting genetic trio-analysis.


\section*{Findings}

Structured reporting and documentation of experimental outcome is required for the successful transfer of knowledge from the research scientist to their peers and to the broader academic community.

Galaxy is a platform that aims to provide complex bioinformatics services and tools in an easy-to-use web-based graphical user interface \cite{goecks2010galaxy,blankenberg2010galaxy,giardine2005galaxy}. The output from these tools can be displayed using built-in Galaxy visualisation applications \cite{goecks}, via specialised visuals implemented as a component in the workflow deployed in Galaxy \cite{cgtag} or by downloading the results and visualising the output with applications external to Galaxy (e.g., Excel, TIBCO spotfire, R, spreadsheet programs, \emph{etc}).

Galaxy has the capacity to track the provenance of the source data, the workflow, as well as the workflow components used to analyse the data. Currently users can share their workflow and results within Galaxy, but do not have access to a simple method to summarise results from multiple tools and/or workflows in an integrated report. To address this issue we have developed iReport, an integrated reporting application that provides users with a flexible means to produce dynamic HTML reports which can be shared with other Galaxy users or downloaded to disk.

Systems used by end-users to deliver graphical output range from open-source applications, such as \emph{Ad Hoc} reports \cite{url-adhoc}, Google charts (and docs) \cite{url-googlecharts} and OpenOffice \cite{url-openoffice}, to commercial applications such as Microsoft Office.
Indeed scientific reporting applications both open-source (Bioconductor \cite{url-bioconductor}, Circos \cite{circos}\cite{url-circos}) and commercial software (e.g., Omniviz \cite{url-omniviz}, Partek \cite{url-partek}) include a multitude of visualisation capabilities with a focus on data reporting and presentation of data in the context of the experimental design and with associated meta-data.
There are some applications, like TIBCO spotfire \cite{url-spotfire}, which are capable of integrating results from multiple sources including associated text and meta-data and other applications which serve as an electronic lab note book (e.g., IDBS \cite{url-idbs}).
Additionally there have been many products developed to address the selection and reporting of variants for pathogenic variant selection including the workflow to identify those variants (e.g., Gensight \cite{url-gensight}, Cartagenia \cite{url-cartagenia}, Clincial Genomicist \cite{clinicalgenomist}).
For data generated in R, dynamic reporting packages such as KnitR \cite{url-knitr}, Sweave \cite{sweave} and R-Markdown \cite{url-rmarkdown}, allow for the integration of data-generating code within the report specification itself. Similar systems exist for other programming languages, for example Tangle \cite{url-tangle} (JavaScript), Active Markdown \cite{url-activemarkdown}(CoffeeScript) or IPython Notebooks \cite{url-ipythonnotebook} (Python).
These are very versatile tools, but require programming knowledge to use effectively. iReport offers an open-source application for both Galaxy and non-Galaxy produced results allowing for the generation of customized integrated reports for any type of project or workflow.
The advantage to Galaxy users is that the output from any application can be included into any report, and that a report template can be reused for other projects.
Also, the report may be securely shared with one or many users of that Galaxy instance or made publicly available. iReports can be completely configured from the tool web interface, and requires no programming or knowledge of the underlying system.

We demonstrate iReportâ€™s utility through an example where a genetic report is generated from outputs of an existing Galaxy next-generation sequencing (NGS) toolkit, CGtag \cite{cgtag}. iReport can also be used as an electronic lab notebook by creating an iReport which links out to various other iReports containing different analysis reports from various samples. It can be also be coupled to output from other Galaxy instances, for example output generated by specialized Galaxy instances such as Confero \cite{confero}, ORIONE \cite{orione}, and Galaxy-P \cite{url-galaxyp}.

\section*{Functionality}
iReport dynamically generates HTML, and employs JavaScript and jQuery to create interactive components, such as searchable, sortable, paginated tables and zoomable images. iReport is ideally suited to use as the final step in a workflow; the pipeline developer configures the report once and end-users are then presented with a templated report each time users run the workflow, while only needing to provide the input files for the pipeline \cite{CGtagpipeline}. iReport can also be used directly by end-users as a means of easily sharing their results with other Galaxy users, or the public via Galaxy's native sharing capabilities.

The generic reporting functionality and usage of iReport is outlined below using an example iReport created for the recent Galaxy Community Conference, which is also available for viewing online \cite{url-ireport-tutorial}. It is followed by an example of a genetic report that can be used for trio analysis, which can easily be modified for any trio reporting or extended to quartets or larger families, also available from our demo galaxy \cite{url-genetic-ireport}.

\section*{iReport Structure}

iReport produced a report webpage consisting of one or more subpages with one or multiple elements included on each subpage. The primary output of iReport is:


\begin{enumerate}
 \item A cover page
  \begin{enumerate}
    \item Title of the report
    \item Cover image
  \end{enumerate}
 \item Main report page consisting of a set of tabs. Each tab consisting of one or more \emph{content items}. Each content item can be one of the following types:
  \begin{enumerate}
    \item Text
    \item Images
    \item Tables
    \item PDF Files
    \item Links
  \end{enumerate}
\end{enumerate}

An iReport tutorial has been developed to demonstrate and explain the functionality of iReport, and is available as a shared history from the CTMM-TraIT public Galaxy instance \cite{url-ireport-tutorial}. The following sections describe each of the components of iReport in more detail.

\subsection*{Cover Page}
The cover page consists of a user-specified title and a cover image. The cover image parameter is optional and when the field is left blank a default image is used (\hyperref[fig:defaultcover]{Figure \ref*{fig:defaultcover}}).
By clicking on the image, or the link above it, the user can access the main report page. There is also a link to download the entire iReport webpage, including all dependency files, for storing or viewing on different systems.

\begin{figure}[h!]
    \includegraphics[width=\textwidth]{chapters/images/iReport/Hiltemann_defaultcover.jpg}
    \caption{Example cover page.Example of a cover page with title \emph{Example Report} and the default cover image. A link to download the entire iReport web page is also provided.}
    \label{fig:defaultcover}
\end{figure}


\subsection*{Main Report Page}
An arbitrary number of tabs may be added via a repeat parameter. Each tab can be labelled with a name specified by the user. An arbitrary number of \emph{content items} may then be added to each tab in a repeat parameter. A type must be specified for each content item (e.g., text, image, table \emph{etc.}), as well as several other parameters depending on the type chosen (\hyperref[fig:wrapper]{Figure \ref*{fig:wrapper}}). Layout is mostly left up to the browser, but users can explicitly add a line-break after each item to force items to appear underneath each other.

% Galaxy wrapper
\begin{figure}[h!]
    \centering
    \includegraphics[scale=0.5]{chapters/images/iReport/Hiltemann_wrapper.jpg}
    \caption{iReport tool wrapper. iReport's tool interface. Minimally a report title and at least 1 tab with 1 content item need to be specified.}
    \label{fig:wrapper}
\end{figure}

\paragraph*{Content item: Text Field}
Text can be entered in a text field in the tool interface, for example to create an introduction paragraph and to give a description of the items on the page. Text is printed verbatim, although a small number of HTML tags are allowed in order to give the user some control over formatting (e.g., \verb+b, i, em, strong, h1-h6+ tags). Text files can also be specified, and the contents of the file will be printed to the screen verbatim.

\paragraph*{Content item: Images}
Many tools produce images as output, which can also be displayed by iReport. Users specify the image file from their Galaxy history, and the desired image size. For images that have been scaled down, an optional jQuery zoom-on-mouseover effect may  be added (\hyperref[fig:imagezoom]{Figure \ref*{fig:imagezoom}})\cite{url-jqueryzoom}. Currently supported image formats are JPG, PNG and SVG.

%% zoomeffect
\begin{figure}[h!]
    \includegraphics[width=\textwidth]{chapters/images/iReport/Hiltemann_zoomeffect.jpg}
    \caption{Zoom effect.Images that have been scaled down can optionally be enhanced with a jQuery zoom-on-mouseover effect. In this example, the bottom image has this effect added, and when the user moves their mouse over the image, a zoomed version of that area of the image is shown. }
    \label{fig:imagezoom}
\end{figure}

\paragraph*{Content item: Tables}
iReport can also display tables. The input must be a tab-delimited file from the users' Galaxy history, and the first nonempty line not starting with a hash symbol (\verb+#+) is assumed to contain the column headers. The jQuery library \emph{DataTables} \cite{url-datatables} is used to create tables which can be searchable, sortable and paginated, if requested by the user. There is an option to create hyperlinks within the columns of a table by providing a column number, a URL prefix and a URL suffix. This is illustrated in \hyperref[fig:collinks]{Figure \ref*{fig:collinks}}, where the first column contains gene names and by including the GeneCards \cite{genecardspaper,url-genecards} URL prefix \verb+http://www.genecards.org/cgi-bin/carddisp.pl?gene=+. This generates a hyperlink to the corresponding GeneCards entry for every item in the column in the table.

%% weblinks from table columns
\begin{figure}[h!]
    \includegraphics[width=\textwidth]{chapters/images/iReport/Hiltemann_columnlinks.jpg}
    \caption{Weblinks from table columns
   A series of web links can be created within a table by specifying a prefix and suffix to be placed before and after each column entry.}
    \label{fig:collinks}
\end{figure}

\paragraph*{Content item: PDF Files}
This is one of the simplest content items. The user provides a PDF file from the Galaxy history, which will be embedded in the page. If the browser does not have the necessary plug-ins installed, a download link for the file will be generated instead (\hyperref[fig:pdfexample]{Figure \ref*{fig:pdfexample}}).

%% PDF example
\begin{figure}[h!]
    \includegraphics[width=\textwidth]{chapters/images/iReport/Hiltemann_PDF.jpg}
    \caption{Embedded PDF files. iReports can also display PDF files. For browsers without PDF plug-in, a download link to the file will be created instead. }
    \label{fig:pdfexample}
\end{figure}

\paragraph*{Content item: Links}
Users can create links to web locations by specifying a URL and a link text. Links to datasets in the history can also be created here by specifying a dataset and a link text. Several tools create archives of files as output (for example a zip file containing the plots for each chromosome). Links to all files contained in an archive can also be created, and will be named with the file names (excluding file extension). Currently the supported archive formats are zip, bz2, tar, gz and tar.gz. An example can be seen in \hyperref[fig:archivelinks]{Figure \ref*{fig:archivelinks}}, where an archive with images was used as input and a series of links to each contained file was created. An option to create a link to an iReport is also present. This allows users to create a kind of electronic lab notebook, by creating an overview of all their samples and linking to one or more iReports for each sample.

%% links from archive
\begin{figure}[h!]
    \includegraphics[width=\textwidth]{chapters/images/iReport/Hiltemann_archivelinks.jpg}
    \caption{Links to all files in an archive.Given an archive of files, iReport can create a series of links to all files contained in the archive. Link texts are the filenames (without file extension).}
    \label{fig:archivelinks}
\end{figure}

\section*{Genetic report for a trio of HapMap individuals}

Accurate, reproducible and traceable reporting is an essentialrequirement to the evaluation of the genetic outcome from any assay \cite{skyline}, including those variations predicted from NGS analysis. Since iReport is capable of including many formats, we have used the outcome from a trio analysis generated from the Complete Genomics \cite{drmanac} NGS platform to demonstrate its utility in representing these data in a user-defined format, which contains the provenance of the underlying analysis. In this example we use a trio of individuals sequenced in the International HapMap Project \cite{hapmap}\cite{hapmap2}, to demonstrate how to select protein affecting candidate variants based on a recessive genetic model.  All data in this example is freely available for download from the Complete Genomics website \cite{url-CGpublicdata}.

This example iReport has one tab devoted to explaining the protocol used (\hyperref[fig:trioscreenshots]{Figure \ref*{fig:trioscreenshots}B}), one tab with circos plots and an explanation of the family structure (\hyperref[fig:trioscreenshots]{Figure \ref*{fig:trioscreenshots}D}), and one tab with tables containing the candidate pathogenic variants determined by the protocol based on a recessive model for selection. This iReport is also available as a published history on the TraIT-CTMM public Galaxy \cite{url-traitgalaxy}.

%% Genetic Report
% file: Hiltemann_geneticreport[A-D].jpg
\begin{figure}[h!]
   \includegraphics[width=\textwidth]{chapters/images/iReport/Hiltemann_geneticreport.jpg}
%    \begin{center}$
%    \begin{array}{cc}
%    \includegraphics[scale=0.08]{chapters/images/iReport/Hiltemann_geneticreportA.jpg}   &
%    \includegraphics[scale=0.08]{chapters/images/iReport/Hiltemann_geneticreportB.jpg}   \\
%    A & B \\
%    \includegraphics[scale=0.08]{chapters/images/iReport/Hiltemann_geneticreportC.jpg}   &
%    \includegraphics[scale=0.08]{chapters/images/iReport/Hiltemann_geneticreportD.jpg}   \\
%    C & D \\
%    \end{array}$
%    \end{center}
    \caption{Example iReport: Genetic Report. Example iReport for Clinical Genetics. A) Cover page with custom image. B) First tab, explaining the protocol used. C) Second tab, tables of candidate pathogenic variants, gene columns linking out to GeneCards. D) Fourth tab showing Circos images and family structure.  }
    \label{fig:trioscreenshots}
\end{figure}

\section*{Conclusions}
iReport is a easy-to-use, flexible tool for generating traceable, standardized reports which are easily shared between users within and across platforms. We have demonstrated that iReport is capable of creating a customised genetics report from results generated within Galaxy and may be shared with collaborators on the same platform, or with the public. Additionally, data or results generated externally can be uploaded into Galaxy and can also be used by iReport. These reports are generated as web pages and may be downloaded in their entirety to be easily shared across systems.

The genetics report presented here represents the bare minimal reporting that is required to summarise the output for a genetic variation analysis. Whilst we used a trio of individuals to demonstrate how to select protein-affecting candidate variants based on a recessive model, any number of model outcomes and other assay results may be included in an iReport.

We developed iReport to simplify reporting and sharing the output from \emph{omics} and non-high throughput assays analysed both in and external to Galaxy. We have also utilised iReport for more complex analysis workflows, such as summarising translational research and diagnostic applications for cancer and immunological research and diagnostics.

\section*{Availability and requirements}
Project name: iReport \\
Project home page: https://github.com/shiltemann/iReport \\
CTMM-TraIT public Galaxy instance: galaxy.ctmm-trait.nl \\
iReport tool shed repository: https://toolshed.g2.bx.psu.edu/view/saskia-hiltemann/ireport \\
\ \\
Operating system(s): Unix-based Operating Systems \\
Programming languages: Bash, Perl, Python \\
Other Requirements: Galaxy \\
License: GNU GPL \\
Any restrictions to use by non-academics: none \\
\ \\
Examples: \\
iReport about iReport published history: \url{http://galaxy.ctmm-trait.nl/u/saskia-hiltemann/h/gcc2014-ireport-about-ireport}, or \url{tinyurl.com/llrzz9w} \\
Clinical Genetics iReport published history: \url{http://galaxy.ctmm-trait.nl/u/andrew-stubbs/h/ireportgeneticreportchr21} \\

%Tool shed repository for iReport: https://toolshed.g2.bx.psu.edu/view/saskia-hiltemann/ireport \\
%GitHub repository: https://github.com/shiltemann/iReport \\

\section*{Availability and supporting data}
The iReport tool, user manual (published page), and example data and histories are available at the CTMM-TraIT Galaxy server \cite{url-traitgalaxy}.

\section*{Abbreviations}
CGtag: Complete Genomics Toolkit and Annotation in a Cloud-based Galaxy. \\
CTMM-TraIT: Center for Translational Molecular Medicine - Translational IT. \\
NGS: Next Generation Sequencing. \\
URL: Uniform Resource Locator. \\


\section*{Acknowledgements}
 This study was performed within the framework of the Center for Translational Molecular Medicine (CTMM). TraIT project (grant 05T-401).

\bibliographystyle{plain}
\bibliography{references}
