\cleartorightpage\begin{savequote}[75mm]
``I was taught that the way of progress was neither swift nor easy.''
\qauthor{Marie Curie}
\end{savequote}

\chapter{Accessible Bioinformatics}\label{chapter:general}
\setcounter{figure}{-1}
\setcounter{table}{-1}
\setcounter{section}{-1}
%\setcounter{enumiv}{-1}
\setcounter{NAT@ctr}{-1}

\begin{figure}[t!]
\includegraphics[height=10em]{frontmatter/images/chapter-header-technical.png}
\end{figure}
\setcounter{figure}{-1}
\setcounter{table}{-1}
\setcounter{section}{-1}

The vast majority of bioinformatics tools are UNIX-based command line applications that are not easy to use for non-bioinformaticians. This chapter describes three resources developed to increase the accessibility of bioinformatics tools and pipelines for non-experts.

The \textbf{Galaxy platform} aims to improve the user-friendliness of bioinformatics tools by providing a web-based graphical user interface to command line UNIX tools. This enables data scientists to run complex bioinformatics analyses needing nothing more than a web browser.

Analysis pipelines often generate a large number of outputs, and with the explosion of data in the next-generation sequencing field, many of these resulting outputs contain millions of lines and can no longer easily be reviewed manually.

Circos is a tool that enables visualisation of high-dimensional data in a circular plot. Circos is extremely flexible but has a high degree of complexity and a steep learning curve. We developed \textbf{Galactic Circos}, a Galaxy version of the Circos tool, to increase its usability for non-bioinformaticians, and to improve interoperability with upstream tools.

Visualisation and summarisation of results has become an integral part of any analysis pipeline. The \textbf{iReport} tool provides generic reporting within the Galaxy framework for easy summarisation of analysis results. This result can help researchers and clinicians quickly interpret their data.


This chapter contains the following sub-chapters:

\begin{enumerate}[label=\ref{chapter:general}.\arabic*]
\itemsep-0.5em
\setcounter{enumi}{-1}
\item \textbf{The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update.} The Galaxy platform is a large project supported by a very big team of developers, scientists and community volunteers. My contributions to this work consisted primarily of the following activities: 1) Developing tools. I contributed over 150 tools to the Galaxy Tool Shed including mothur, cgatools, iReport, Circos, and many more, as well as a number of workflows. Furthermore, as a member of the IUC (Intergalactic Utilities Commission) I was involved in developing best-practice guidelines for tool development. I was also involved in the reviewing of new tools and updates to existing tools, to ensure they adhere to these guidelines. Furthermore, I was actively involved in the community-building efforts around the tools. 2) Training. As described in more detail in the next chapter, I co-founded the Galaxy Training Materials project, an effort to centralize and FAIR-ify training materials that use the Galaxy platform. 3) I created several \emph{interactive environments} in Galaxy; unlike most tools, these special container-based tools allow for interactive exploration of data. I created interactive environments for Phinch (visualization of microbiome data), Ethercalc (a spreadsheet application), and contributed to the Rstudio interactive environment. I also made some (minor) contributions to the infrastructure supporting interactive environments in Galaxy. 4) I created a connector between the Owncloud data sharing platform (\url{https://owncloud.com/}) and Galaxy, adding a "Send to Galaxy" button on datasets in Owncloud, and vice versa. 5) I've co-created Galaxy \emph{flavours}, most notably for metagenomics. These docker-based topical Galaxy instances provide ready-to-deploy Galaxy containers optimized for a specific research domain. 6) As an administrator of a (small) public Galaxy server, I've contributed bug fixes to the Galaxy base code over the years as well. Furthermore, I often provide feedback about the requirements of researchers in order to guide the roadmap of Galaxy developments and ensure these are in line with the needs of the community.


\item \textbf{Galactic Circos: User-friendly creation of Circos Plots within the Galaxy platform.} Circos is a very popular tool for visualisation of genome-scale data, especially in publications. While Circos is incredibly powerful and flexible, it is also complex to the point of requiring bioinformatics expertise to use. Helena Rasche and I created a Galaxy port of this tool, so that Circos plots can be configured by users via the browser. Due to the complexity of Circos, this became probably the single-most complex Galaxy tool wrapper as well. While not exposing the full power of Circos, it strikes a good balance between versatility and complexity, between flexibility and user-friendliness. Because of the complexity of this tool, we also created detailed training materials as part of the publication in order to increase the accessibility of the tool.


\item \textbf{iReport: A generalised Galaxy solution for integrated experimental reporting.} I developed this Galaxy tool as a way to offer generic, configurable analysis results reporting within Galaxy. These reports are crucial for clinical applications where the most important analysis results need to be presented in a succinct and shareable report. Note that this paper refers to the CTMM-TraIT Tool Shed, which has since been taken offline. However, The iReport tool is now available in the main Galaxy Tool Shed. Furthermore, in the years since I wrote iReport, the Galaxy team has recognized the importance of this tool, and has worked towards incorporating this functionality into the Galaxy code base itself. Termed \emph{Galaxy workflow reports}, an iReport-like results report can now be configured into the Galaxy workflow definition itself. While iReport is still available from the Tool Shed and can still be used, I have halted any further iReport development in favour of contributing to these Galaxy workflow reports and optimizing their utility in a research and clinical setting.
\end{enumerate}



