\begin{savequote}[75mm]
"Humans are allergic to change. They love to say, 'We've always done it this way.' I try to fight that. That's why I have a clock on my wall that runs counter-clockwise."
\qauthor{Grace Hopper}
\end{savequote}

\chapter{Discussion}
\label{discussion}
\setcounter{figure}{-1}
\setcounter{table}{-1}
\setcounter{section}{-1}
\setcounter{NAT@ctr}{-1}

Since the completion of the human reference genome in 2003, the field of molecular biology has been transformed almost beyond recognition, and the field of bioinformatics has evolved from a niche discipline to an integral part of every biomolecular research question. Where just a couple decades ago most genomic data analyses could largely be performed by hand, nowadays they often require a supercomputer. Programming knowledge is required to run such analyses, but biologists are not typically trained in these skills, and have thus become reliant on bioinformaticians to carry out this work for them. Similarly, bioinformaticians often lack the increasingly complex biological knowledge required to fully interpret the results. Biologist and bioinformatician must therefore work together closely, and each must be trained in the other discipline sufficiently enough to be aware of any factors that might influence analysis or interpretation. Furthermore, data is being generated at an exponential rate, while bioinformaticians are not, and one way to address this gap is to empower researchers to run their own day-to-day analyses after analysis pipelines are fully developed and validated.


\section{Galaxy for open and accessible research}
The Galaxy project~\cite{TODO} is a framework that enables such empowerment of researchers to run complex data analysis without programming expertise. The Galaxy project is free and open-source and community-driven, solliciting feedback and code contributions from its user community to improve and evolve along with the ever-changing landscape that is bioinformatics. \hyperref[chapter:galaxy]{\textbf{Chapter~\ref{chapter:general}}} outlines the ongoing development in the Galaxy framework. Not only are new tools continually integrated by the community, new components are incorporated into the Galaxy code base and existing features improved to evolve and scale with user requirements. On the user-facing side, improvements were made to assist with the handling of large datasets (collections, rule-based uploader) to match the exponential rate of data generation. Accessibility was improved by the integration of a new help forum and the development of a central training materials repository (\textbf{Chapter~\ref{chapter:training}}).


\section{Training}
Training is an essential component in the dissemination of accessible bioinformatics workflows. Training is needed both for researchers looking to analyze their data, as well as for the bioinformaticians developing the tools and workflows. Galaxy is especially well-suited for bioinformatics training because it provides a layer of abstraction, allowing trainees to focus on the bioinformatics \emph{concepts} rather than the nitty-gritty details about the commandline interface. For example, when teaching sequence mapping, the focus in teaching can lie purely on the different paremeters and their effects on downstream analysis, whereas when teaching the same steps on the commandline, participants would be learning two things at once; how the unix commandline works as well as mapping to a reference genome, increasing the cognitive load of students, which may hamper learning \cite{paas2003cognitive}.

In Chapter \ref{chapter:training} we created a framework for collaborative easy all the good adjectives delivery of bioinformatics training using the Galaxy platform.

<we focus on ease of contribution>

<some stats on contributors and new tutos etc>

Since the initial publication in Chapter \ref{chapter:training} we have continued development of the GTN framework. As Galaxy evolves, so must the associated tutorials, and where Galaxy is expanding beyond bioinformatics and is also used in fields such as natural language processing and cheminformatics, so do we notice



\section{Visualisation and Reporting}

iReport

Galaxy is highly flexible, but this comes hand-in-hand with complexity. This trade-off is acceptable for researchers still developing their pipelines, but for clinicians who have fixed and validated workflows, the flexibility is not longer required, sterker nog undesirable, and the complexity becomes a hurdle. To this end, simplified forntends for Galaxy are desired. Bioblend. Galaksio. (myFAIR?)


\section{Genome Sequencing}

long read sequencing

single cell sequencing

\section{Cancer Analysis}

\section{Microbiota Analysis}


\section{Future Perspectives}
\subsection{The Bio: Futuromics}

\subsubsection{Single-cell Sequencing}

\subsection{The Informatics}
\begin{comment}
- bigger better more RAM/CPU to keep up with the ever increasing data sizes, cloudification
- make it usable - the biologists have the knowledge for interpretation, software should be
  usable by them, not just bioinformaticians, Galaxy/docker/conda etc helps with this
- make it maintainable as developers by adhering to coding best-practices and through the
  use of continuous integration strategies>
- training is vital, the more complex these data and methods get, the harder it will be to
  draw accurate conclusions>
- quantum computing? blockchain? the future or just hype?
\end{comment}

\begin{comment}
\section{Academia}

The \textit{publish or perish} attitude in academia is bad; too many tools not maintained after first publication because a bioinformaticians worth seems to be determined solely by the number of first- or last-authorships of publications and the amount of funds obtained through grants. This leads a to huge replication of efforts, with many groups developing software to solve the exact same problem just because either the existing tools were inaccessible, or analysis not reproducible and not maintained, in large part because most bioinformaticians working in academia do not get time allocated for such tasks.

\begin{verbatim}
- Open Science, open software, open data, open all the things!

- Q: in academia often judged on number of papers you publish, but are citations each year
     to existing papers also valued? this would encourage maintainance of tools more, if
     50 new citations to your tool that year is considered just as good as a new first/last
     authorship?

- Paywalled journals need to die
  - even abused by Scott Pruitt to screw over the environment:

- Peer review process should also be open (anonymous if desired by referees, but open)
  - allow for multiple updates to paper? like code in github, if you've updated the tools you
    write about, or analysed additional samples, or even if just URLs change, make it easier
    to submit new versions, with open peer review, enable re-reviews of each new version,
    --> like F1000 model
  - or blind peer review, referees and journtal editors don't know the authors when making
    their decision so cannot be biased (some of worst tools I've encountered are from
    hot-shot names in big fancy journals)

- Do we even need journals?

- science communication: Every publication should be accompanied by a short description for
  layman readers, media often grossly misinterprets results, do better about science
  communication
\end{verbatim}

just testing a citation:~\cite{hiltemann2014ireport}.
\end{comment}

\bibliographystyle{ieeetr}
\bibliography{references}
