\begin{savequote}[75mm]
``Humans are allergic to change. They love to say, `We've always done it this way.'' I try to fight that. That's why I have a clock on my wall that runs counter-clockwise.''
\qauthor{Grace Hopper}
\end{savequote}

\chapter{Discussion}\label{discussion}
\setcounter{figure}{-1}
\setcounter{table}{-1}
\setcounter{section}{-1}
\setcounter{NAT@ctr}{-1}

Since the completion of the human reference genome in 2003, the field of molecular biology has been transformed almost beyond recognition, and the field of bioinformatics has evolved from a niche discipline to an integral part of every biomolecular research question. Where just a couple decades ago most genomic data analyses could largely be performed by hand, nowadays they often require a supercomputer. Programming knowledge is required to run such analyses, but biologists are not typically trained in these skills, and have thus become reliant on bioinformaticians to carry out this work for them. Similarly, bioinformaticians often lack the increasingly complex biological knowledge required to fully interpret the results. Biologist and bioinformatician must therefore work together closely, and each must be trained in the other discipline sufficiently enough to be aware of any factors that might influence analysis or interpretation. Furthermore, data is being generated at an exponential rate, while bioinformaticians are not, and one way to address this gap is to empower researchers to run their own day-to-day analyses after analysis pipelines are fully developed and validated.


\section{Galaxy for open and accessible research}
The Galaxy project~\cite{TODO} is a framework that enables such empowerment of researchers to run complex data analysis without programming expertise. The Galaxy project is free and open-source and community-driven, solliciting feedback and code contributions from its user community to improve and evolve along with the ever-changing landscape that is bioinformatics. \hyperref[chapter:galaxy]{\textbf{Chapter~\ref{chapter:general}}} outlines the ongoing development in the Galaxy framework. Not only are new tools continually integrated by the community, new components are incorporated into the Galaxy code base and existing features improved to evolve and scale with user requirements. On the user-facing side, improvements were made to assist with the handling of large datasets (collections, rule-based uploader) to match the exponential rate of data generation. Accessibility was improved by the integration of a new help forum and the development of a central training materials repository (\textbf{Chapter~\ref{chapter:training}}).


\section{Training}
Training is an essential component in the dissemination of accessible bioinformatics tools and workflows. Training is needed both for researchers looking to analyze their data, as well as for the bioinformaticians developing the tools and workflows. Galaxy is especially well-suited for bioinformatics training because it provides a layer of abstraction, allowing trainees to focus on the bioinformatics \emph{concepts} rather than the nitty-gritty details about the UNIX commandline interface. Take for example a workshop on sequence mapping; when using Galaxy for teaching, the focus can lie purely on the different parameters and their effects on downstream analysis, whereas when teaching the same steps on the commandline, participants would have to learn new details about the UNIX commandline in addition to read mapping. Such an approad increases the cognitive load of students, which may hamper learning \cite{paas2003cognitive}.

The Galaxy Training Network (GTN) is a loosely define group of instructors around the world who use Galaxy for training purposes, but there was little coordination between the different instructors in terms of materials used, and thus a lot of duplication of effort. Therefore, we set out to create a central repository of Galaxy training materials and a set of best-practice guidelines which could be used by anybody in the community. Any central solution would have to be a collaborative effort if it was to be sustainable and maintainable. In Chapter~\ref{chapter:training} we describe the collaborative web framework for delivery of bioinformatics training using the Galaxy platform that we created in response to this need.

We focused on creating a fully open and transparent framework that is accessible and easy to use for both learners and trainers. All development happens on GitHub, where anybody may suggest additions and changes, and any such proposed changes are thoroughly tested using Travis continuous integration system \cite{travis-ci} to ensure functionality and adherence to guidelines. The proposed changes or additions to tutorials or the framework are then reviewed by one or more of the volunteers from the community, and changes may be requested. Once approved, the code is merged into the main code base, and the new website is automatically built and deployed by Travis and GitHub. The materials themselves are based around \emph{research stories}; usually the recreation of results described in a published paper. This gives users the confidence that the tools and pipelines they are learning are practically useful and publication-level quality, as well providing them with the opportunity to explore the science and informatics in the paper in full detail. Furthermore, since publication of our work, a number of publications of scientific papers have included Galaxy training materials for readers and as a form of documentation \cite{TODO}.

One of the main challenges in designing this framework, was to allow easy contributions from instructors, without the need for any web development knowledge. To this end, we used Jekyll templating \cite{jekyll}, which allows tutorials to be written in the simple and accessible markup language Markdown \cite{markdown}, which is then automatically converted to a web page. Analogous to how Galaxy allows scientists to run analyses while being abstracted away from the implementation layer of the tools, so does jekyll allow instructors to create web pages for their tutorials without being concerned with the web application layer.

A further challenge was to enable the materials to be usable both by instructors during workshops, as by individuals learning on their own. This is accomplished by including all materials instructors might provide during a workshop in the GTN training materials framework. This includes slides as well as hand-on materials, further reading suggestions, and an automatically created and updated list of Galaxy servers which meet the requirements to run a given tutorial.

\begin{comment}
<closer integration with galaxy servers>

<feedback, planemo, cofests, dashboard, instructor topic, levels, translations, tess search, curricula, toc>

2019: 4 new topics (metabolomics, computational chemistry, data manipulation, ui and features), 66 new tutorials
\end{comment}

Since the initial publication of Chapter~\ref{chapter:training}, we have continued active development of the GTN framework and training materials. As Galaxy evolves, so must the associated tutorials; where Galaxy is expanding beyond bioinformatics and is now also being used in fields such as natural language processing and computational chemistry, so do we notice a steady expansion of topics and tutorials contributed by the community. In the year after publication, we saw 6 new topics added, 66 new tutorials, and the number of contributors grew from 64 to 137. Where the focus initially lay with the end-users of the tutorials, in the year after publication, the focus shifted to supporting of instructors. This was done through the additions of tutorials for contributors and instructors about how to contribute training materials and teach workshops.

In order to support this growing and diverse community of learners and trainers, constant maintenance and development is required. The main challenge in the coming years will be the community management; creating and sustaining a close-knit community of Galaxy users and instructors so that the project can survive even when its original developers have moved on.



\section{Visualisation and Reporting}

Galaxy is highly flexible, but this comes hand-in-hand with complexity. This trade-off is acceptable for researchers still developing their pipelines, but for clinicians who have fixed and validated workflows, the flexibility is not longer required, and in many cases even undesirable. At this point the relative complexity of the Galaxy interface to novice users may become a hurdle.

One of the main deficits of Galaxy in its current form, is the lack of an appealing system of results reporting. Datasets produced from workflows must be individually viewed by the user. While Galaxy sports a plugin system for visualisation of individual datasets, a generic reporting tool for displaying a set of output datasets together does not exist. To this end, we developed iReport (Chapter \ref{}), a fully customizable Galaxy tool that generates a webpage that can display any number of workflow outputs. In this manner, end-users need only to open a single workflow output to get a full overview of results. The workflow developer can create an iReport fully tailored to the needs of their users, including the ability to add links to datasets and external resources, create searchable and sortable tables, add images and custom text, and to divide content into different pages. In this manner, clinicians and other end users are able to run workflows and view results with minimal instruction and knowledge of the pipelines and Galaxy interface.

Going even further, by using the Galaxy API, simplified web frontends for Galaxy may be created, such that only features needed by the users are exposed. In the case of clinical applications, this provides the additional benefit of protecting user from inadvertently altering any workflow parameters. An example of such a simplified front-end is Galaksio \cite{galaksio} or IRIDA \cite{}. Users log in to a custom webpage, are given a list of workflow to choose from, and are asked to provide input datasets. While Galaxy is used to run the analysis, the users never leave their web environment, and everything is handled by the front-end. In order to fully


\section{Genome Sequencing}

The concepts described in the previous sections were applied to two separate use cases. Chapter \ref{chapter:} describes the creation of analysis tools and pipelines for (prostate) cancer analysis, and in Chapter \ref{chapter:} Galaxy-based analysis pipelines were built and tested for microbiota profiling. While from a biological perspective these application fields are quite different, from a bioinformatics perspective they are very similar. Both involve genome sequencing data, which have to be quality checked and mapped to a (set of) reference sequence(s). After these common steps the pipelines start to diverge more significantly; where the cancer pipelines focus on variant detection, microbiota profiling is concerned with the identification of the microbial entities present. Both have the ultimate end-goal of improving diagnosis and treatment of patients.

both need reporting, possibly front end, viz, long reads to improve quality

\subsection{Cancer Analysis}

\subsection{Microbiota Analysis}


\section{Future Perspectives}

Galaxy: reporting, bigger data, other application, IDC, shared reference data,

\subsection{The Bio: Futuromics}
single cells, real-time, bigger data

microbiota: long reads to improve shotgun analysis, or 16S higher accuracy

\subsection{The Informatics}

Galaxy in the clinic: reporting, frontends

Open Science! Open Science! Open Science!

\bibliographystyle{ieeetr}
\bibliography{references}
