\cleartorightpage
\setcounter{NAT@ctr}{-1}
\chapter*{}

\begin{figure}[t!]
\centering
\includegraphics[height=10em]{frontmatter/images/chapter-header-tools.png}
\end{figure}
\vspace{-4cm}

\phantomsection\addcontentsline{toc}{section}{The Informatics: GmT}
\articletitle{Galaxy mothur Toolset (GmT): a user-friendly application for 16S rRNA gene sequencing analysis using mothur.}
Saskia Hiltemann\textsuperscript{\ref{affil:emc-bioinf},*},
Stefan Boers\textsuperscript{\ref{affil:emc-microbiology},*},
Peter van der spek \textsuperscript{\ref{affil:emc-bioinf}},
Ruud Jansen\textsuperscript{\ref{affil:streeklabhaarlem}},
John Hays\textsuperscript{\ref{affil:emc-microbiology}},
Andrew Stubbs\textsuperscript{\ref{affil:emc-bioinf}}

\small
\begin{enumerate}
\itemsep-0.5em
\item Department of Bioinformatics, Erasmus Medical Center, Rotterdam, The Netherlands.\label{affil:emc-bioinf}
\item Department of Medical Microbiology and Infectious Diseases, Erasmus Medical Centre, Rotterdam, The Netherlands\label{affil:emc-microbiology}
\item Department of Molecular Biology, Regional Laboratory of Public Health Kennemerland, Haarlem, The Netherlands.\label{affil:streeklabhaarlem}
\end{enumerate}

{\color{chaptergrey}{Published in:}} \emph{GigaScience}, 2019 Feb 1;8(2):giy166 \\
{\color{chaptergrey}{DOI:}} \url{https://doi.org/10.1093/gigascience/giy166} \\
{\color{chaptergrey}{*:}} Saskia D. Hiltemann and Stefan A. Boers contributed equally to this work.


\section*{Abstract}
\textbf{Background} The determination of microbial communities using the mothur tool suite (\url{https://www.mothur.org}) is well established. However, mothur requires bioinformatics-based proficiency in order to perform calculations via the command-line. Galaxy is a project dedicated to providing a user-friendly web interface for such command-line tools (\url{https://galaxyproject.org/}).

\textbf{Results:} We have integrated the full set of 125+ mothur tools into Galaxy as the Galaxy mothur Toolset (GmT) and provided a set of workflows to perform end-to-end 16S rRNA gene analyses and integrate with third-party visualization and reporting tools.  We demonstrate the utility of GmT by analysing the mothur MiSeq standard operating procedure (SOP) data set (\url{https://www.mothur.org/wiki/MiSeq\_SOP}).

\textbf{Conclusions:} GmT is available from the Galaxy Tool Shed, and a workflow definition file and full Galaxy training manual for the mothur SOP have been created. A Docker image with a fully configured GmT Galaxy is also available

\textbf{Keywords:} Microbial classification; 16S rRNA gene sequence analysis; mothur

\section*{Findings}

\subsection*{Introduction}
 rRNA gene profiling analysis can be achieved using an extensive array of sophisticated software including mothur~\cite{schloss2009introducing}, QIIME~\cite{caporaso2010qiime}, MG-RAST~\cite{glass2010using}, and many more~\cite{oulas2015metagenomics}. Whilst some of these applications have a graphical user interface (GUI) to provide access to these technologies for the research scientist, their use remains complex for non-bioinformaticians. In this respect, the Galaxy project~\cite{afgan2016galaxy} was developed in order to simplify the use of complex command-line software tools. Galaxy offers extensive support for both 16S rRNA gene-based and broader metagenomic analyses, with over 100 tools in the metagenomics section of the Galaxy tool shed, including QIIME~\cite{caporaso2010qiime}, KRONA~\cite{ondov2011interactive}, PyNAST~\cite{caporaso2009pynast}, PICRUSt~\cite{langille2013predictive}, Kraken~\cite{wood2014kraken}, MetaPhlAn2~\cite{truong2015metaphlan2}, HUMAnN2~\cite{abubucker2012metabolic}, PrinSEQ~\cite{schmieder2011quality}, Nonpareil~\cite{rodriguez2013nonpareil}, Vegan~\cite{dixon2003vegan}, and many more.

Mothur is an open-source application that was designed as a single piece of software capable of analysing and comparing microbial communities from 16S rRNA gene data derived from next-generation sequencing (NGS). The creators of mothur did not only provide an extensive set of tools, but also a collection of standard operating procedures (SOPs) that detail the recommended analytical protocol for different types of input data.

The latest version of mothur consists of over 125 components, lending it great flexibility, but at the same time, great complexity. To address this challenge, we have integrated the full set of 125+ mothur components into Galaxy that are collectively called the `Galaxy mothur Toolset (GmT)`. To simplify usage of GmT we provide the full workflow definition files, usage of which shields the end-user from the full complexities of the analysis. By simultaneously providing access to all the individual components present in mothur as separate tools, expert users and bioinformaticians retain the ability to utilize the full flexibility of mothur by creating custom workflows or by modifying or extending our workflows to fit their use-case.

GmT also leverages Galaxy's collections framework to enable easy analysis of large numbers (many thousands) of samples at once. Many mothur components support parallel computing, and the Galaxy tools will utilize the maximum amount of processing power alloted to them by the instance administrator. As part of GmT, datatypes were also contributed to the Galaxy core codebase to facilitate the handling of mothur-specific datatypes within Galaxy. Furthermore, a Galaxy data manager was also created for the automatic installation and configuration of reference datasets utilized by the mothur tool suite. And lastly, a Galaxy Interactive environment (GIE)~\cite{helena2015galaxy} for Phinch~\cite{bik2014phinch} was also developed~\cite{phinchGIE}.

GmT includes tools to produce standard file formats, such as the BIOM format~\cite{biom-format} to facilitate interoperability with these downstream analysis components. Where no clear file standards exists, GmT provides custom tools for conversion of mothur datatypes to other tools (e.g.\ the taxonomy-2-krona tool). This allows for integration with third party tools such as PICRUSt for prediction of functional content, or visualisation tools such as Phinch, KRONA, and certain QIIME components. The mothur tools also natively support incorporation of some 3rd party analysis tools, such as UCHIME and ChimeraSlayer for chimera detection or VSEARCH for clustering, which are also available in GmT.

The Galaxy Training Network (GTN) is a network of people and groups that present Galaxy and Galaxy-based training around the world. The GTN has created a central repository~\cite{gtn-repo} for Galaxy training materials. In order to further facilitate the use of GmT to end-users, we have contributed training materials to the GTN that illustrate how to run mothur's MiSeq SOP within Galaxy~\cite{gmt-training}. This work has also been incorporated in a larger-scale framework to easily and quickly explore microbiota data in a reproducible and transparent environment~\cite{batut2017asaim}.

\subsection*{Purpose of this work}
The work performed and described in this technical note has four objectives. First to provide end-users and bioinformaticians with easy access to all the mothur tools as the GmT. Second is to provide open-access online training material to demonstrate/complete the mothur SOP in Galaxy.  Third is to deliver an end-to-end workflow for the mothur SOP in Galaxy that is available for upload to any Galaxy that has the GmT installed.  Fourth is to provide a summarization of results in a web report using the iReport Galaxy tool~\cite{hiltemann2014ireport}. Our aim is to provide 16S rRNA gene NGS analysis tools and awareness on how to use them in a format that supports FAIR data principles~\cite{wilkinson2016fair}.

\subsection*{Worked Example}
To illustrate the utility of our toolkit, we present results on example data below. GmT is designed to take short-read 16S rRNA gene NGS data as input and to output a dynamic web report for prokaryotic taxonomical classification using the Galaxy platform.  A GmT workflow follows essentially a four-step process:

\begin{enumerate}
\item \textbf{Data upload}.
The Galaxy platform provides the users with standard data upload functionality for single and multi-sample datasets.

\item \textbf{Collection Creation}.
For multi-sample and/or paired-end datasets a Galaxy collection must be created in the Galaxy interface. Here datasets can also be assigned to groups. Galaxy will make intelligent suggestions for pairings of datasets based on the file names.

\item \textbf{16S rRNA gene analysis}. Mothur has been wrapped as a tool suite in Galaxy. Required steps included for a full \textit{end-to-end} 16S rRNA gene sequencing analysis consist of read-pair merging (mothur command: make.contigs), trimming of primer sequences (trim.seqs), additional quality control (screen.seqs), alignment of sequences to a (customized) reference alignment (align.seqs, screen.seqs), removal of chimeric sequences (chimera.uchime), classifying sequences using a Bayesian classifier in combination with a reference database such as SILVA or GreenGenes (classify.seqs), and clustering of sequences into operational taxonomic units (OTUs) at a predefined percentage - usually 97 percent - of similarity (dist.seqs, cluster, and classify.otu) (\hyperref[end-to-end]{Figure~\ref{end-to-end}}).

\item \textbf{Experimental Summary and Reporting}.\ iReport in combination with KRONA is used to deliver an HTML report in Galaxy~\cite{ondov2011interactive}. The iReport consists of multiple tabs to group results topically (e.g.\ taxonomy, rarefaction, diversity, quality control) and is highly customizable and easily tailored to an end-user's specific use-case. The entire report may be downloaded from the Galaxy interface to be viewed or shared offline.

%The interactive iReport consists of three tabs. The first tab (Taxonomy) visualizes and lists the resultant microbiota profiles with a direct linkage of each representative sequence derived from each OTU detected to the NCBI database that allows for additional BLASTN searches. The second tab (Diversity) summarizes the results of 3 diversity calculators (Chao1, Shannon, and Simpson). The third tab (Quality Control) provides an extensive overview of the quality control measurements during the bioinformatics analysis. All results tables presented within the iReport, as well as all representative sequences derived from each OTU measured, can be downloaded that allows user to (further) process or check the quality of the 16S rRNA gene sequencing analysis using their own software programs.

To compare the output from a single experiment or across multiple experiments we utilized Phinch~\cite{bik2014phinch}, a dynamic web application which uses BIOM-formatted files to explore and analyse biological patterns in 16S rRNA gene NGS datasets.

\end{enumerate}

\begin{figure}[t!]
\centering
\includegraphics[scale=0.5]{chapters/images/gmt/Figure1.png}
\caption{Conceptual view of the GmT mothur MiSeq SOP pipeline.}\label{end-to-end}
\end{figure}

\begin{figure}[t!]
\centering
\includegraphics[scale=0.5]{chapters/images/gmt/ireport.png}
\caption{\textbf{Example iReport.} This web report contains the interactive KRONA visualization, the (multi-sample) OTU table, rarefaction plots, diversity calculations, differential abundance analysis, and an extensive overview of the quality control measurements taken during the analysis.\@ iReports are highly customizable and can be easily tailored to fit specific use-cases and end-user needs.}\label{ireport}
\end{figure}


\section*{Methods}

\subsection*{Handling large datasets}
Large-scale analyses have become the norm in the field, both large in disk space as in the number of files, and this can pose a challenge for analysis. For large files, Galaxy offers the option of uploading via FTP rather than web transfer. The introduction of the concept of "collections" in Galaxy has enabled users to analyze datasets consisting of a large number of files (>100K) as easily as they would a single file.

\subsection*{Galaxy mothur Toolset}
Many mothur components support parallelization, and our Galaxy wrappers will run these components with the maximum number of CPUs allotted to them by the Galaxy administrator. In order to diagnose potential failures, Galaxy outputs the full standard and error logs, which the users can inspect. Furthermore, we have contributed mothur datatype definitions to the Galaxy core code, meaning that the users will be protected from inputting the wrong datasets and thus reduce the number of errors they will make with the tools. All tools in GmT use only conda dependencies, making their installation in Galaxy a painless experience that requires nothing more than a single press of a button.

The mothur tool wrappers have been submitted to the Intergalactic Utilities Commission (IUC) tool repository~\cite{iuc-repo} and are available from the Galaxy Tool Shed~\cite{toolshed}. The IUC is a group of community members dedicated to developing and upholding Galaxy tool development best practices and guidelines, thus by contributing our tools to this repo we ensure that the tools will be well-maintained. A metagenomics Galaxy flavour~\cite{metagenomics-flavour} is available which contains all components presented here. The full mothur suite has also been installed to Galaxy's main server.~\cite{usegalaxy}.

\subsection*{KRONA visualization}
KRONA~\cite{ondov2011interactive} is a data viewer which provides the ability to interactively explore hierarchical data. A Galaxy KRONA wrapper that works directly on mothur data formats was developed for this project.

\subsection*{Phinch visualization}
Galaxy offers integration with Phinch~\cite{bik2014phinch} BIOM format viewer in two ways; as a Galaxy interactive environment (GIE) developed in the context of this project~\cite{phinchGIE}, and more recently also as an external display application hosted by the Galaxy team.

\subsection*{iReport summarization}
To facilitate the evaluation of 16S rRNA gene sequencing analysis results, integration with the iReport~\cite{hiltemann2014ireport} tool are also provided. This tool creates a web report to present the analysis results in an organized fashion and provides links to external resources such as BLAST searches (\hyperref[ireport]{Figure~\ref{ireport}}).

\section*{Availability of source code and requirements}

\begin{itemize}
\item Project name: Galaxy mothur Toolset (GMT)
\item Project home page: \url{https://github.com/erasmusmc-bioinformatics/galaxy-mothur-toolset}
\item Toolshed repository: \url{https://toolshed.g2.bx.psu.edu/view/iuc/suite_mothur/768c2e48b706}
\item Training manual: \url{https://galaxyproject.github.io/training-material}
\item GmT Docker image: \url{https://quay.io/shiltemann/galaxy-mothur-toolset:16.07}
\item Galaxy Metagenomics
Docker Flavour (Docker): \url{https://quay.io/repository/shiltemann/galaxy-metagenomics}, \url{https://github.com/shiltemann/galaxy-metagenomics}
\item Phinch interactive environment: \url{https://github.com/shiltemann/phinch-galaxy-ie}
\item Operating system: Unix (Platform independent with Docker)
\item License: GNU GPL v3
\end{itemize}

\section*{Availability of supporting data and materials}

The data presented here to illustrate our work is the same data used in the training manual, and is available from Zenodo~\cite{zenodo-data}

\section*{Declarations}

\subsection*{List of abbreviations}

\begin{itemize}
\item \textbf{GIE:} Galaxy interactive environment
\item \textbf{GmT:} Galaxy mothur Toolset
\item \textbf{GTN:} Galaxy Training Network
\item \textbf{GUI:} Graphical user interface
\item \textbf{SOP:} Standard Operating Procedure
\end{itemize}

\subsection*{Competing Interests}
The authors declare that they have no competing interests.

\subsection*{Funding}
This work has received funding from the European Union’s Seventh Framework Programme for Health under grant agreement number 602860 (TAILORED-Treatment; www.tailored-treatment.eu).

\subsection*{Author's Contributions}

SH developed the Galaxy tool wrappers and Phinch interactive environment. SB validated the analysis pipelines. All authors contributed to the manuscript text and approve its contents.

\section*{Acknowledgements}

The authors would like to thank Jim Johnson and the many other contributors and reviewers of the mothur tool wrappers, including everybody who contributed to the development of these tools within the context of the Galaxy metagenomics contribution fest organized by the Galaxy community's Intergalactic Utilities Commission (IUC), a group of community members dedicated to developing and upholding Galaxy tool development best practices and guidelines~\cite{iuc}. We would also like to thank the Galaxy Training Network for providing the infrastructure and valuable feedback to share our training materials.


\bibliographystyle{ieeetr}
\bibliography{references}
