%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty		       %%
%%									   %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    



% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}



\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{CGtag: Complete Genomics Toolkit and Annotation in a Cloud-based Galaxy}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Saskia Hiltemann\correspondingauthor$^{1,2}$%
       \email{Saskia Hiltemann\correspondingauthor - s.hiltemann@erasmusmc.nl}%
      \and
         Hailiang Mei$^3$%
         \email{Hailiang Mei - hailiang.mei@nbic.nl}
       \and
         Mattias de Hollander$^4$%
         \email{Mattias de Hollander - m.dehollander@nioo.knaw.nl}%
       \and
         Ivo Palli$^1$%
         \email{Ivo Palli - i.palli@erasmusmc.nl}%
       \and
         Peter van der Spek$^1$%
         \email{Peter van der Spek - p.vanderspek@erasmusmc.nl}%
       \and
         Guido Jenster$^2$ %
         \email{Guido Jenster - g.jenster@erasmusmc.nl}%
       and
         Andrew Stubbs$^1$%
         \email{Andrew Stubbs - a.stubbs.erasmusmc.nl}%	  
      }
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address{%
    \iid(1)Department of Bioinformatics, Erasmus MC, Rotterdam, %
         The Netherlands\\
    \iid(2)Department of Urology, Erasmus MC, Rotterdam, %
         The Netherlands\\    
    \iid(3)Netherlands Bioinformatics Center (NBIC), Nijmegen, The Netherlands \\
    \iid(4)Department of Microbial Ecology, Netherlands Institute of Ecology (NIOO-KNAW), Wageningen, The Netherlands 	
}%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
        % Do not use inserted blank lines (ie \\) until main body of text.
        \paragraph*{Background:} Complete Genomics (CG) provides an open-source suite of command-line tools for the analysis of their CG formatted mapped sequencing files.  Determination of for instance the functional impact of detected variants requires annotation with various databases that often require command-line and/or programming experience and thus limit their use to the average research scientist.  We have therefore implemented this CG toolkit, together with a number of annotation and visualisation tools in Galaxy called CGtag (Complete Genomics Toolkit and Annotation in a Cloud-based Galaxy).  
        \paragraph*{Findings:} In order to provide research scientists with web-based, simple and accurate analytical and visualisation applications for the selection of candidate mutations from Complete Genomics data, we have implemented the open-source Complete Genomics tool set, CGATools, in Galaxy.  In addition we implemented some of the most popular command-line annotation and visualisation tools to allow research scientists to select candidate pathological mutations (SNV, and indels).  Furthermore, we have developed a Cloud-based public Galaxy instance to host the CGtag toolkit and other associated modules.
        \paragraph*{Conclusions:} CGtag provides a user-friendly interface to all research scientists wishing to select candidate variants from CG or other next-generation sequencing platforms’ data. By using a Cloud-based infrastructure, we can also assure sufficient and on-demand computation and storage resourses to handle the analysis tasks. The tools are freely available for use from an NBIC (The Netherlands Bioinformatics Center) Cloud-based Galaxy instance \cite{url-nbicgalaxy}, or can be installed to a local Galaxy via the NBIC toolshed \cite{url-nbictoolshed}
        
        \paragraph*{Keywords:} Complete Genomics, Next Generation Sequencing, Genetic variation, Pathogenic gene selection.
\end{abstract}



\ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%




%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Findings }
\subsection*{Background}
Complete Genomics supplies results for whole-genome NGS mapped to a user-defined genome \cite{ma} and additional open-source tools \cite{url-cgatools} for further characterization of the sequenced genomes.  Whilst these tools are open-source and available for download and use on the command-line they are not amenable for the research scientist to use from their desktop and require scripting skills to link these tools together with other applications to successfully prioritise candidate pathogenic genes based on these NGS results.  To address this issue, we implemented the Complete Genomics Analysis Toolkit (CGATools), including several functional annotation and visualization tools in a Cloud-enabled instance of Galaxy \cite{drmanac}.  Galaxy offers a web-based graphical user interface to command-line tools, and allows for the graphical construction of complex workflows, will automatically keep track of the analysis history, and permits easy sharing and publishing of data and/or workflows with other users. Furthermore, Galaxy is an extensible platform, nearly any software tool may be integrated into Galaxy, and there is an active community of users and developers ensuring the latest tools are made available for use in Galaxy through the Galaxy Toolshed.

This implementation of the CG Toolkit in a Galaxy environment simplifies the analysis of genomes via the Galaxy GUI and the Cloud resource ensures that sufficient computing power is available for the analysis.  The inherent functionality in Galaxy of CGtag delivers customizable user-defined workflows by the research scientist and not only by the bioinformatician.   

\subsection*{Variant Detection}
The Complete Genomics Analysis Toolkit (CGATools) is an open-source project to provide tools for downstream analysis of Complete Genomics data, and may be downloaded from their repository \cite{url-cgatools}. These tools must be run from the command-line and therefore are not accessible to all users. To remedy this, Complete Genomics also provide Galaxy tool wrappers for many of the CGAtools, which can be downloaded from the Main Galaxy tool repository (toolshed) \cite{url-toolshed}. However, these Galaxy tools still need to be installed on the users’ local Galaxy instance before they can be utilized. We have now made these tools available on a public server \cite{url-nbicgalaxy}, and have added Galaxy tools for those CGAtools that were not provided by Complete Genomics e.g. Junctions2Events, makeVCF (Table 1).  The use of the tools in Table 1 have previously been outlined \cite{nieminen}, using a combination of listvariants and testvariants or calldiff to determine candidate pathogenic SNVs, indels and subs in a selected genome as compared with on or more reference genomes or as part of a trio based genetic analysis \cite{nieminen}.  The Varfilter may be used to select those variants which have a high confidence based on the underlying sequence reads as specified as VQHIGH and SNPDiff can then be used to determine concordance of the NGS results with those of an orthogonal SNV detection platform such as an Affymetrix or Illumina SNP array.  The JunctionDiff and Junction2Events tools are used to select fusion events and candidate fusion genes based on quality of the discordant reads used to detect the structural variation event \cite{ifuse}.

\subsection*{Functional Annotation Tools}
To provide users with enhanced filtering capabilities, we have integrated several command-line annotation tools in this NBIC Galaxy instance. ANNOVAR \cite{annovar} is a command-line tool used to functionally annotate genetic variants. We provide a Galaxy tool wrapper for ANNOVAR. This tool will take a list of variants as input and provide gene annotation, amino acid change annotation, SIFT scores, PolyPhen scores, LRT scores, MutationTaster scores, PhyloP conservation scores, GERP++ conservation scores, DGV variant annotation, dbSNP identifiers, 1000 Genomes Project allele frequencies, NHLBI-ESP 6500 exome project allele frequencies and other information.  We have implemented this tool to accept VCF (v4) files, Complete Genomics varfiles or CG-derived tab-separated files using the CG 0-based half-open coordinate system, or lastly, the standard ANNOVAR input format consisting of tab-separated lists of variants using the 1-based coordinate system.  This tool will output the original file columns, followed by additional ANNOVAR columns.  To supplement ANNOVAR, Condel (CONsensus DELeteriousness) \cite{condel} has been included to calculate the deleterious score associated of missense SNVs and the impact of non-synonymous SNVs on protein function. Condel integrates the outputs of 2 tools: SIFT, Polyphen2, to calculate a weighted average of the scores (WAS) of these tools. Condel can optionally incorporate the output of a third tool, MutationAssessor, which is also included in this Galaxy instance. Mutation Assessor \cite{mutass} is a web-based tool providing predictions of the functional impact of amino-acid substitutions in proteins, such as mutations discovered in cancer or missense polymorphisms. The MutationAssessor database is accessed through a REST API.  In order not to overload the server, queries are limited to 3 per second, so when dealing with a long list of variants, some pre-filtering is recommended. The functional annotation provided by ANNOVAR including the addition of multiple versions of dbSNP, the variants provided by Complete Genomics Public data from unrelated individuals only \cite{url-cgftp} and 31 genomes from Huvariome \cite{huvariome} are available in this Galaxy instance. Huvariome provides the user with additional whole genome variant calls for and for those regions which difficult to sequence and can retrieve the weighted allele frequency for each base in the human genome \cite{huvariome}.   

\subsection*{Visualisation Tools}
A generic genomic data plotter tool based on GNUplot is available, which takes as input a tab-delimited file of format chr–start-end–value, and will output either a single chromosome plot, an overview of all chromosome plots in a single image,  or a sub-region of a chromosome defined by the user. Additionally, the tool has the option of plotting input from a second file in the same image, which is useful for tumour-normal comparison (Figure 1). B-allele frequency (BAF) is used to determine whether the structural variation junction is homo- or heterozygous. When the data is in the right format, the generic plotter tool can be used to visualize the BAF, and we have also implemented a plot tool to display allele frequencies directly from a CG masterVar file, again with the capability of displaying single-chromosome plots, all chromosomes in a single image, or custom defined regions (Figure 1). The current Complete Genomics  Analysis Pipeline (CGAP v2.0) delivers Circos \cite{url-circos} visualisations with each genome that is sequenced and the code used to generate these images have been made freely available for download \cite{url-cgcircos}.  We have modified this code and implemented Galaxy tool to allow for the generation of these images for samples sequenced on earlier CG Analysis pipelines (before v2.0) which utilises the junctions file, masterVar file, CNV details and CNV segments files to generate the standard CG Circos report. 

To support fusion-gene analysis we have created a custom Circos tool which using CG files, CG junctions file and CG varfile for NGS, and the results from SNP arrays analysis, specifically the B-allele frequency (BAF) and copy number variation (CNV) files.  The output is either a whole-genome plot, per-chromosome plots, a single image containing all the per-chromosome plots together, or a plot of a custom region defined by the user (e.g. a plot showing just chromosomes 3, 5, and X, or a plot showing a specific range within a single chromosome).  Additionally the user can select an “impacted genes” track for the per-chromosome plots, which will print the names of the genes impacted by SV events along the outer edge of the image (Figure 2).  This custom Circos script is capable of using fusion gene detection results generated from the Illumina platform with the fusion genes detected by an application such as FusionMap \cite{fusionmap} and reported in FusionMap report format, a tab-delimited format similar to that from delivered by Complete Genomics. 

In addition to these tools within Galaxy, structural variation files processed using CGtag may be exported to our previously described fusion gene prioritization tool, iFUSE \cite{url-ifuse} to identify candidate fusion genes and display their representative DNA, RNA and protein sequence. 

\subsection*{Auxiliary Tools}
Our suite of tools also includes several auxiliary tools supplied by CG but not available from the Galaxy tool shed which offer the user several file format conversion tools (Table 1) which enable users to connect the output from the CGATools analysis to other analytical or annotation workflows by means of standard file formats (e.g. FASTA, VCF). 

\subsection*{CLOUD Implementation}
NBIC Galaxy is hosted at a high performance computing cloud system operated by SURFsara \cite{url-surfsara}. This HPC cloud consists of 19 fast servers with 608 CPUs and almost 5TB of memory. The NBIC Galaxy that operates in this HPC cloud is implemented using the Cloudman framework \cite{afgan} and its adapted version supporting OpenNebula Cloud environment. 
The advantage of using the Cloudman framework to build NBIC Galaxy is mainly two-fold, firstly  Cloudman provides a set of complete scripts to automatically install tools and datasets on a virtual machine image. The installed tools include Galaxy system itself and all its dependencies. These dependencies include webserver (nginx), database (postgres), cluster job scheduler (SGE), and common NGS tools, such as bowtie, BWA, samtools, etc. The installed datasets include most of the common reference genomes (hg18, hg19, mm9, etc) and their tool specific index files. Thus, the end product of running Cloudman installation script is a fully functional NBIC Galaxy system operating in the HPC Cloud. 

The second contribution of Cloudman to our NBIC Galaxy system is its ability to set up a flexible virtual cluster and the auto-scaling support. The previous NBIC Galaxy was hosted on a dedicate physical server with rather limit resource (4 CPU, 32G memory). Due to this resource limitation, our NBIC Galaxy was never promoted to be a real data analysis server to handle production level of NGS datasets. On the other hand, because of the sporadic nature of user access, the server was mostly on idle during its 2 years lifespan. Moving to Cloud resolved both issues. The current NBIC Galaxy operates on top of a virtual cluster. This virtual cluster contains one head node and a number of worker nodes. These nodes are all virtual machines that are built using the machine image generated by the Cloudman script. When there is just a minimal usage, the cluster will only contain one head node. Once a significant load occurs due to training courses or production level data analysis, the virtual cluster can automatically scale up itself. More worker nodes will be added dynamically to this virtual cluster to boost the capacity of NBIC Galaxy. Once the load decreases, the virtual cluster can scale down again to operate with only limited number of nodes.

The use of shared resource does have drawback as well. We have experienced a more obvious I/O bottleneck in the Cloud based NBIC Galaxy compared to the previous system that ran in a physical machine. In the HPC Cloud, storage is provided through a NFS system instead of a local hard disk. When more concurrent Cloud users are using the Cloud resource, we observe the extra job time caused by I/O delays. However, we argue that this issue is far outweighs the benefit of having a dynamic virtual cluster support to the NBIC Galaxy.
    
%%%%%%%%%%%%%%%%%%%%%%
\section*{Availability and requirments}
Project Name: CGtag: Complete Genomics Toolkit and Annotation in a Cloud-based Galaxy\\
Project home page: http://galaxy-demo.trait-ctmm.cloudlet.sara.nl\\
Operating system: Linux (Galaxy and CGtag)\\
Programming language: Python (Galaxy and CGtag), R (CGtag), Bash (CGTag) \\
Other requirements: Circos \cite{url-circos}, GNUplot \cite{url-gnuplot}, Complete Genomics open source Toolset \cite{url-cgatools}  and dependencies therein);  see documentation for a comprehensive list of optional dependencies, based on workflow requirements.\\
License: GPL v3\\
Any restrictions to use by non-academics: No\\


\section*{List of abbreviations}
NBIC: The Netherlands Bioinformatics Center \\
CG: Complete Genomics \\
CGtag: Complete Genomics Toolkit and Annotation in a Cloud-based Galaxy\\
SV: Structural Variation\\
SNV: Single Nucleotide Variation\\
BAF: B-allele Frequency

\section*{Competing interests}
None declared.


%%%%%%%%%%%%%%%%%%
\section*{Submission of supporting data to the GigaScience database, GigaDB}
None. Tools are available for use at a public NBIC cloud-based Galaxy instance \cite{url-nbicgalaxy}, or can be installed to a local Galaxy instance via the toolshed \cite{url-nbictoolshed}. 
Some public test data from Complete Genomics is available from the NBIC Galaxy as a shared dataset, or directly from CG via ftp \cite{url-cgftp}.
    
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}
SH, GJ, HM and AS contributed to the design and coordination of CGtag and manuscript preparation. SH, MdH, IP and HM contributed to implementing CGtag.  SH, GJ, PvdS and AS contributed to testing of CGtag. MdH and HM implemented Galaxy Cloudman for the SURFsara/Big-grid HPC cloud. All authors read and approved the final manuscript. 

    

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgements}
\ifthenelse{\boolean{publ}}{\small}{}
This study was performed within the framework of CTMM, the Center for Translational Molecular Medicine. TraIT project (grant 05T-401). \\

This work was sponsored by the BiG Grid project for the use of the computing and storage facilities, with financial support from the Netherlandse Organisatie voor Wetenschappelijk Onderzoek (Netherlands Organisation for Scientific Research, NWO)


 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%              
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %% 
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %% 
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
 \bibliographystyle{bmc_article}  % Style BST file
 \bibliography{CGtag.bib} }     % Bibliography file (usually '*.bib' ) 

%%%%%%%%%%%

\ifthenelse{\boolean{publ}}{\end{multicols}}{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \subsection*{Figure 1 - Generic Genomic Data Plotting Tool}
      Output from our generic genomic data plotter used to plot B-allele frequency from Illumina 1M SNParray data. Plot with two tracks; tumour (red) and normal (black). Output can be (top) a whole genome overview (shown here in part), or (middle) a single chromosome, or (bottom) a subregion of a chromosome defined by the user (here chr16, 60MB-end).  Many parameters such as the colour and sizes of the data points may be adjusted by the user as required.

  \subsection*{Figure 2 - Circos Integrative Plot Tool}
      Circos plots for (left) whole genome, (middle) overview or all chromosomes in single images, and (right) for a single chromosome.  Each chromosome is represented in the outer ring and then from outer to inner rings represent copy number variation (with regions of gain depicted in green and loss in red), B-allele frequency, SNP density and the intra- and interchromosomal rearrangements are on the inside and depicted in black and red lines, respectively.  Impacted genes track (red gene symbols) are displayed outside the outer chromosome ring and only on the single chromosome plot.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
  \subsection*{Table 1 - CGATools}
    Overview of CG Tools available in Galaxy (All are latest release of version 1.7)
    should go. \par \mbox{}
    \par
    \mbox{
      \begin{tabular}{l|l|l}
        
        \textbf{Function} & \textbf{CGATool}  & \textbf{Description} \\ \hline         
        Variant Detection & ListVariants & Lists the non-redundant set of small variations  \\ 
	                  &              & found in an arbitrary number of genomes. \\ \hline
        Variant Detection & TestVariants & Determine which variants are found in which   \\ 
                          &              & genomes given the results of ListVariants. \\ \hline
        Variant Detection & CallDiff     & Compares two variant files to determine where \\
                          &              & and how the genomes differ. \\ \hline
        Variant Detection & VarFilter    & Copies the input varfile or masterVar file, \\
                          &              & applying filters. \\ \hline	
        Variant Detection & JunctionDiff & Reports difference between junction calls \\
                          &              & of CG junction files. \\ \hline
        Variant Detection & Junctions2Events & Groups and annotate related junctions. \\ \hline
        Quality Control   & SNPDiff      & Compares genotype calls to CG variant files. \\ \hline
        File Merge        & Join         & Merge tow tab-delimited files based on equal \\
                          &              & field or overlapping regions. \\ \hline
        Sequence retrieval& DecodeCRR    & Retrieve sequences from a CRR file for a given \\
                          &              & range of a chromosome. \\ \hline
        File Conversion   & mkVCF        & Converts CG variant and/or junction files to VCF. \\ \hline
        File Conversion   & generateMasterVar & Converts a varfile to a on-line-per-locus format. \\ \hline
        File Conversion   & fasta-2-crr  & Converts fasta sequences into a single reference crr file. \\ \hline
        File Conversion   & crr-2-fasta  & Converts crr file to fasta sequence. \\ \hline
        File Conversion   & TestVariants2VCF & Converts output of TestVariants tool to VCF. \\ 
        
      \end{tabular}
      }
  
\end{bmcformat}
\end{document}







